tests/_cli/markdown_data/dataflow.md.txt

```diff
@@ -0,0 +1,429 @@
+---
+title: Dataflow
+marimo-version: 0.0.0
+---
+
+# How marimo notebooks run
+
+Reactive execution is based on a single rule: when a cell is run, all other
+cells that reference any of the global variables it defines run
+automatically.
+
+To provide reactive execution, marimo creates a dataflow graph out of your
+cells.
+## References and definitions
+
+A marimo notebook is a directed acyclic graph in which nodes represent
+cells and edges represent data dependencies. marimo creates this graph by
+analyzing each cell (without running it) to determine its
+
+- references ("refs*), the global variables it reads but doesn't define;
+- definitions ("defs"), the global variables it defines.
+
+There is an edge from one cell to another if the latter cell references any
+global variables defined by the former cell.
+
+The rule for reactive execution can be restated in terms of the graph: when
+a cell is run, its descendants are run automatically.
+### Example
+
+The next four cells plot a sine wave with a given period and amplitude.
+Each cell is labeled with its refs and defs.
+```{marimo}
```

Comment:
what are your thoughts about a new line between "markdown to code"?

Reply:
Agreed